### PR TITLE
fix: handle negative model predictions in visualizations

### PR DESCRIPTION
**Pull request by rmnmllr**
related to issue https://github.com/scikit-hep/cabinetry/issues/388.

Tried `python -m pytest`, did not pass all tests even before modifying anything. No additional errors though after the fix and tested with samples mantioned in issue https://github.com/scikit-hep/cabinetry/issues/388.

### CODE REVIEW

**Comment by rmnmllr** [comment]
@alexander-held I'm not sure why the python tests fail, pytest gives an error
```
>       assert "predicted yield is zero in 1 bin(s), excluded from ratio plot" in [
            rec.message for rec in caplog.records
        ]
E       AssertionError: assert 'predicted yield is zero in 1 bin(s), excluded from ratio plot' in ['predicted yield is zero in 1 bin(s),            excluded from ratio plot']
```
but this is not something I touched.

File: src/cabinetry/visualize/plot_model.py [hunk 1]
```diff
@@ -145,10 +145,16 @@ def data_mc(
     n_zero_pred = sum(total_yield == 0.0)  # number of bins with zero predicted yields
     if n_zero_pred > 0:
         log.warning(
-            f"predicted yield is zero in {n_zero_pred} bin(s), excluded from ratio plot"
+            f"predicted yield is zero in {n_zero_pred} bin(s),\
+            excluded from ratio plot"
```
[the remark this answers is not in the document]
Comment (rmnmllr) [reply src/cabinetry/visualize/plot_model.py:149]: Argh of course, will fix that hopefully tomorrow!

**Comment by rmnmllr** [comment]
> Thanks a lot for preparing this! I am afraid it may take me until next week to more carefully look at this, sorry for the slow feedback here.

No worries, I'm glad I can contribute in some way!

**Comment by alexander-held** [comment]
Hi @rmnmllr, I added tests to this to get this ready for inclusion in the next version of `cabinetry` that I intend to release with #390 also being completed. Thanks a lot for preparing this!

```
* catch negative total model prediction in visualize.plot_model.data_mc
* handle negative nominal template prediction in visualize.plot_model.templates
```

**Pull request merged**